tests/utils.py

```diff
@@ -197,3 +197,40 @@ def create_proxy_block_file(blocked_domains: List[str]):
         file.flush()
 
         yield file.name
+
+
+@contextlib.contextmanager
+def slow_app(host: str, port: int):
```

Comment:
There's probably a way we could achieve this as an ASGI app?

It's not super clear to me what in this code makes the app "slow". I assume it's that that we `sleep(0.5)` and the test uses a `connect` timeout of 0.1?

Reply:
> There's probably a way we could achieve this as an ASGI app?

…But `hypercorn` only allows us to test that on 3.7+. Hmm, that's actually a real bummer that we can't use Hypercorn to test against arbitrary apps on 3.6.